kairos-cli/src/crypto/signer.rs

```diff
@@ -0,0 +1,38 @@
+use std::path::Path;
+
+use super::private_key::CasperPrivateKey;
+use super::public_key::CasperPublicKey;
+use crate::crypto::error::CryptoError;
+use casper_types::bytesrepr::ToBytes;
+use casper_types::{crypto, ErrorExt, PublicKey};
+
+pub struct CasperSigner {
+    secret_key: CasperPrivateKey,
+    pub public_key: CasperPublicKey,
```

Comment:
Why do we need to have a public key here?

Reply:
Public key is necessary for [signing data with Casper signer](https://docs.rs/casper-types/4.0.1/casper_types/crypto/fn.sign.html):

```rs
pub fn sign<T: AsRef<[u8]>>(
    message: T,
    secret_key: &SecretKey,
    public_key: &PublicKey
) -> Signature
```

Since we have user's private key, we can derive public key - we could do that every time `sign_message()` is called, but I thought storing it might come handy soon.

Reply:
Ahh... don't use that signer, it has an unused public key for some reason.

Use this instead:

```rust
pub fn sign<T: AsRef<[u8]>>(secret_key: &SecretKey, message: T) -> Signature {
    match secret_key {
        SecretKey::System => Signature::System,
        SecretKey::Ed25519(secret_key) => {
            let signature = secret_key.sign(message.as_ref());
            Signature::Ed25519(signature)
        }
        SecretKey::Secp256k1(secret_key) => {
            let signature = secret_key
                .try_sign(message.as_ref())
                .expect("should create signature");
            Signature::Secp256k1(signature)
        }
        _ => panic!("SecretKey is marked as non-exhaustive, but this should never happen"),
    }
}
```